### PR TITLE
feat(cartitem): add productURLPath prop to allow more specific URLs

### LINKED ISSUE #357: Switching billing address options doesn't trigger onReadyForSaveChange
Type: **minor**

**Describe the bug**
On Stripe checkout, the Save button remains disabled when user clicks "Use a different billing address" then switches back to "Same as shipping address."

**To Reproduce**
1. Create a cart. Checkout.
2. Fill in the shipping address and pick a shipping option.
3. In the payment form, fill in all required credit card info, then select "Use a different billing address". See that the Save button is disabled.
4. Now select "Same as shipping address." See that the Save button is still disabled.

**Expected behavior**
When user clicks "Use a different billing address" then switches back to "Same as shipping address", the Save button should be enabled.

### PR DESCRIPTION
**Pull request by machikoyasuda**
Resolves #348 
Impact: **minor**  
Type: **feature|bugfix**

## Bug
Part of the fix for this bug https://github.com/reactioncommerce/reaction-next-starterkit/issues/385

## CartItems, CartItem, CartItemDetail
- Adds a `productURLPath` prop to `CartItems`, passed down to`CartItem` and `CartItemDetail`
- A product's URL in CartItems can now be structured like this: `localhost:3000/product-url-path/product-slug` using `localhost:3000{productURLPath}{productSlug}` props.
- A product's productURLPath can be set at a higher-level at `CartItems`, `MiniCart` or `FinalReviewCheckoutAction`, or it can be set at the individual `CartItem` level.

## MiniCart, FinalReviewCheckoutAction
- Add an example in the docs for each with `productURLPath` passed down into `CartItems`

## Testing

Test passing `productURLPath` in all of these components:

- CartItem: https://deploy-preview-348--stoic-hodgkin-c0179e.netlify.com/#!/CartItem
- CartItems: https://deploy-preview-348--stoic-hodgkin-c0179e.netlify.com/#!/CartItems
- CartItemDetail: https://deploy-preview-348--stoic-hodgkin-c0179e.netlify.com/#!/CartItemDetail
- MiniCart: https://deploy-preview-348--stoic-hodgkin-c0179e.netlify.com/#!/MiniCart
- FinalReviewCheckoutAction: https://deploy-preview-348--stoic-hodgkin-c0179e.netlify.com/#!/FinalReviewCheckoutAction

### CODE REVIEW

**Comment by rc-publisher** [comment]
🚀 Preview deployed

Built with commit 8e3bfbe62afd9d1ee6ec991278a24db9a8dda845

https://deploy-preview-348--stoic-hodgkin-c0179e.netlify.com

File: package/src/components/FinalReviewCheckoutAction/v1/FinalReviewCheckoutAction.js [hunk 1]
```diff
@@ -139,7 +139,8 @@ class FinalReviewCheckoutAction extends Component {
         CartSummary
       },
       label,
-      stepNumber
+      stepNumber,
+      ...props
```
Comment (machikoyasuda) [inline package/src/components/FinalReviewCheckoutAction/v1/FinalReviewCheckoutAction.js]: @nnnnat Should I add `productURLPath` explicitly as a prop to `FinalReviewCheckoutAction`, or is passing it through implicitly though `...props` okay?

File: package/src/components/FinalReviewCheckoutAction/v1/FinalReviewCheckoutAction.md [hunk 2]
```diff
@@ -55,6 +55,7 @@ const checkoutSummary = {
   onReadyForSaveChange={isReady}
   checkoutSummary={checkoutSummary}
   stepNumber={4}
+  productURLPath="/product"
```
Comment (nnnnat) [inline package/src/components/FinalReviewCheckoutAction/v1/FinalReviewCheckoutAction.md]: Missing props doc

**Comment by machikoyasuda** [comment]
@nnnnat Updated MiniCart and FinalReviewCheckoutAction to add `productURLPath` prop docs

**Pull request merged**